### PR TITLE
Add ResNet implementation with pre-trained weights

### PR DESCRIPTION
**Pull request by laggui**
Issue #11 

This PR adds a Burn implementation for the ResNet family of computer vision models. Implementation and weights are based on [torchvision](https://pytorch.org/vision/stable/models/resnet.html).

# TO-DO
- [x] ResNet implementation for resnet-{18, 34, 50, 101, 152}
- [x] Inference example
- [x] Use PyTorchFileRecorder to import weights
- [ ] Switch to next Burn release version
  - The `Cargo.toml` file currently points to a local workspace copy of Burn since parsing the resnet weights `with_key_remap` requires a fix that came in after the `0.12.1` release and a fix on Candle's side to [handle module parameters](https://github.com/huggingface/candle/pull/1681) with pickle
- [ ] Automatic download of pre-trained weights (need to re-export [`download_file_as_bytes`](https://github.com/tracel-ai/burn/blob/main/burn-dataset/src/vision/downloader.rs) for common use in burn)

### CODE REVIEW

File: resnet-burn/Cargo.toml [hunk 1]
```diff
@@ -11,16 +11,13 @@ default = ["burn/default"]
...
 ] } # alloc is for no_std, derive is needed
 
 [dev-dependencies]
-burn = { version = "0.12.1", default-features = false, features = ["ndarray"]}
-burn-import = "0.12.1"
-# burn = { path = "../../burn/burn", default-features = false}
-# burn-import = {path = "../../burn/burn-import"}
-# image = { version = "0.24.7", features = ["png", "jpeg"] }
\ No newline at end of file
+burn = { path = "../../burn/burn", default-features = false} # local burn to include the with_key_remap chain fix
```
Comment (nathanielsimard) [inline resnet-burn/Cargo.toml:21]: I don't think we need to specify burn in both dev and normal dependencies.

**Review by nathanielsimard — CHANGES_REQUESTED** [review]

File: resnet-burn/Cargo.toml [hunk 2]
```diff
@@ -11,16 +11,13 @@ default = ["burn/default"]
 
 [dependencies]
 # Note: default-features = false is needed to disable std
-burn = { version = "0.12.1", default-features = false}
-# burn = { path = "../../burn/burn", default-features = false}
+burn = { path = "../../burn/burn", default-features = false}
```
[the remark this answers is not in the document]
Comment (laggui) [reply resnet-burn/Cargo.toml:14]: Agreed! It's still the last item in my TODO list to finalize the PR but the rest was ready for review.

Will put a specific git hash for now.

File: resnet-burn/Cargo.toml (hunk 1 above)
Comment (laggui) [reply resnet-burn/Cargo.toml:21]: Whoops, you're right.

**Comment by laggui** [comment]
Addressed the changes for the current version.

Would be great to have a version that includes the fixes but that also depends on Candle releasing a new version with the module parameters fix for pickle.

**Review by nathanielsimard — APPROVED** [review]
LGTM

**Pull request merged**